src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java

```diff
@@ -74,10 +76,110 @@ public void testChromeLinux() {
+    assertEquals("chrome", result.get(UserParamsCollator.USER_BROWSER_TYPE));
+    assertEquals("mac", result.get(UserParamsCollator.USER_PLATFORM));
+    assertEquals("99", result.get(UserParamsCollator.USER_BROWSER_VERSION));
+  }
+
+  @Test
+  public void testClientHintsInEdge() {
+    ClientHints clientHints = new ClientHints();
+    clientHints.setMobile(true);
+    clientHints.setPlatform("Windows");
+    clientHints.setArchitecture("x86");
+    clientHints.setPlatformVersion("11.3.1");
+    clientHints.setBrowserUAWithFullVersion(
+        "\" Not A;Brand\";v=\"99.0.0.0\", \"Chromium\";v=\"99.0.1150.46\", \"Microsoft Edge\";v=\"102.0.1150.46\"");
+    String userAgent = "";
+
+    TargetDeliveryRequest request = requestWithClientHintsAndUA(clientHints, userAgent);
+    Map<String, Object> result = collator.collateParams(request, pageLoad);
+    assertEquals("edge", result.get(UserParamsCollator.USER_BROWSER_TYPE));
+    assertEquals("windows", result.get(UserParamsCollator.USER_PLATFORM));
+    assertEquals("102", result.get(UserParamsCollator.USER_BROWSER_VERSION));
+  }
+
+  @Test
+  public void testBrowserVersionFromWhenClientHintAbsent() {
+    ClientHints clientHints = new ClientHints();
+    clientHints.setMobile(true);
+    clientHints.setPlatform("macOS");
+    clientHints.setArchitecture("x86");
+    clientHints.setPlatformVersion("11.3.1");
+    String userAgent =
+        "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.109 Safari/537.36";
+
+    TargetDeliveryRequest request = requestWithClientHintsAndUA(clientHints, userAgent);
+    Map<String, Object> result = collator.collateParams(request, pageLoad);
+    assertEquals("chrome", result.get(UserParamsCollator.USER_BROWSER_TYPE));
+    assertEquals("mac", result.get(UserParamsCollator.USER_PLATFORM));
+    assertEquals("98", result.get(UserParamsCollator.USER_BROWSER_VERSION));
+  }
+
+  @Test
+  public void testClientHintsPrecedenceOverUserAgent() {
+    ClientHints clientHints = new ClientHints();
+    clientHints.setMobile(true);
+    clientHints.setPlatform("macOS");
+    clientHints.setArchitecture("x86");
+    clientHints.setPlatformVersion("11.3.1");
+    clientHints.setBrowserUAWithFullVersion(
+        "\" Not A;Brand\";v=\"99.0.0.0\", \"Chromium\";v=\"99.0.4844.83\", \"Google Chrome\";v=\"99.0.4844.83\"");
+    String userAgent =
+        "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.109 Safari/537.36";
+
+    TargetDeliveryRequest request = requestWithClientHintsAndUA(clientHints, userAgent);
+    Map<String, Object> result = collator.collateParams(request, pageLoad);
+    assertEquals("chrome", result.get(UserParamsCollator.USER_BROWSER_TYPE));
+    assertEquals("mac", result.get(UserParamsCollator.USER_PLATFORM));
+    assertNotEquals("98", result.get(UserParamsCollator.USER_BROWSER_VERSION));
```

Comment:
@shandilya3 - I'm confused why this is 98 if we are using the client hint and not user agent.

Reply:
this is a negative test/assertNotEquals.  It should be 99 for assertEquals positive test.

Reply:
Ah, my mistake for missing that.  In that case I think this test is effectively the same as `testClientHintsBrowserUAFullVersionInChrome`.  Maybe we can get rid of one of them.

Reply:
If we expect the value to be 99, wouldn't it be more accurate to just `assertEqual("99",...)`.  I'm confused too.

Reply:
**assertNotEquals**("98", result.get(UserParamsCollator.USER_BROWSER_VERSION));

Reply:
Let me remove this test, it's causing confusion.  😃